lnrpc/walletrpc/walletkit.proto

```diff
@@ -0,0 +1,157 @@
+syntax = "proto3";
+
+import "github.com/lightningnetwork/lnd/lnrpc/signrpc/signer.proto";
+
+package walletrpc;
+
+message KeyReq {
+    /**
+    Is the key finger print of the root pubkey that this request is targeting.
+    This allows the WalletKit to possibly serve out keys for multiple HD chains
+    via public derivation.
+    */
+    int32 key_finger_print = 1;
+
+    /**
+    The target key family to derive a key from. In other contexts, this is
+    known as the "account".
+    */
+    int32 key_family = 2;
+}
+
+message AddrRequest{
+    // No fields, as we always give out a p2wkh address.
+}
+message AddrResponse {
+    /**
+    The address encoded using a bech32 format.
+    */
+    string addr = 1;
+}
+
+message Transaction {
+    /**
+    The raw serialized transaction.
+    */
+    bytes tx_hex = 1;
+}
+message PublishResponse {
+    /**
+    If blank, then no error occurred and the transaction was successfully
+    published. If not the empty string, then a string representation of the
+    broadcast error.
+
+    TODO(roasbeef): map to a proper enum type
+    */
+    string publish_error = 1;
+}
+
+message SpendReport {
+    /**
+    If true, then this outpoint has been spent a confirmed or unconfirmed
+    transaction within the internal transaction store of the wallet.
+    */
+    bool has_spend = 1;
+
+    /**
+    The raw transaction that spends the target outpoint. Note that this will
+    only be populated if the above field returns true.
+    */
+    bytes raw_tx_hex = 2;
+}
+
+message OutPoint {
+    /**
+    The txid of the transaction that produced this output.
+    */
+    bytes txid = 1;
+
+    /**
+    The index of the output within the transaction above that is being
+    referenced.
+    */
+    int32 index = 2;
+}
+
+message SendOutputsRequest {
+    /**
+    The number of satoshis per kilo weight that should be used when crafting
+    this transaction.
+    */
+    int64 sat_per_kw = 1;
+
+    /**
+    A slice of the outputs that should be created in the transaction produced.
+    */
+    repeated signrpc.TxOut outputs = 2;
+}
+message SendOutputsResponse {
+    /**
+    The txid of the transaction sent out on the network.
+    */
+    bytes txid = 1;
```

Comment:
Can this return the mapping of outputs to tx out indices?

Reply:
Proposal:
https://github.com/btcsuite/btcwallet/pull/568/files

Just returning the change index won't work because of the output randomization that is happening.

Reply:
No longer needed now that we return the entire transaction.